src/network/networkd-dhcp4.c

```diff
@@ -1181,7 +1181,7 @@ static int dhcp4_handler(sd_dhcp_client *client, int event, void *userdata) {
                                 }
 
                                 r = sd_ipv4ll_start(link->ipv4ll);
-                                if (r < 0)
+                                if (r < 0 && r != -ESTALE) /* On exit, we cannot and should not start sd-ipv4ll. */
```

Comment:
As far as I can see with this patch applied the dhcp client sends release messages when networkd is stopped:
```
    1   0.000000      0.0.0.0 → 255.255.255.255 DHCP 330 DHCP Discover - Transaction ID 0xe3d84750
    2   0.000327 192.168.125.193 → 192.168.125.203 DHCP 327 DHCP ACK      - Transaction ID 0xe3d84750
    3   5.890927 192.168.125.203 → 192.168.125.193 DHCP 307 DHCP Release  - Transaction ID 0xe3d84750
```
The crash is gone too.